Feat: PR deployments

nbey opened the pull request.

nbey committed:
feat: implement docker builds

- create gatekeeper-composite habitat plan

nbey committed:
feat: deploy PR builds

- generate charts via helm

nbey committed:
fix: Dockerfile location

nbey committed:
fix: set env variables

nbey committed:
fix: helm deployment template

- remove namespace definition
- loop through hab env values

nbey committed:
docs: pr deployments

nbey committed:
chore: remove deprecated secrets

nbey committed:
fix: ingress annotations for pr deployments

nbey committed:
chore: re-enable docker build

themightychris commented on k8s/charts/deployment/Chart.yaml:
```diff
@@ -0,0 +1,23 @@
+apiVersion: v2
+name: gatekeeper-deployment
+description: A Helm chart for Gatekeeper Kubernetes PR Deployments
```
just a chart for deploying gatekeeper

themightychris commented on k8s/charts/deployment/templates/deployment.yaml:
```diff
@@ -0,0 +1,99 @@
+        env:
+          - name: HAB_GATEKEEPER_COMPOSITE
+            value: |
+            {{- if .Values.hab.gatekeeper_composite.mysql }}
+              [services.mysql]
+              {{- range $key, $value := .Values.hab.gatekeeper_composite.mysql }}
+              {{ $key }} = {{ $value | quote }}
+              {{- end }}
+            {{- end }}
```
there isn't anything but `pkg_ident` that you would configure on the composite at this level, so we can cut down the dynamicism and just write the one value

themightychris commented on k8s/charts/deployment/templates/deployment.yaml:
```diff
@@ -0,0 +1,99 @@
+    {{- include "gatekeeper-deployment.labels" . | nindent 4 }}
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    cert-manager.io/cluster-issuer: letsencrypt-prod
```
make this a value

themightychris reviewed (changes requested):
@nbey couple minor tweaks but overall: perfect!

nbey committed:
chore: minor tweaks

- use helm template for  cert manager annotations
- update helm chart description
- simplify `pkg_ident` helm config
